src/connectionController.ts

```diff
@@ -116,142 +122,188 @@ export default class ConnectionController {
     this._telemetryService = telemetryService;
   }
 
-  async _migratePreviouslySavedConnection(
-    savedConnectionInfo: StoreConnectionInfo
-  ): Promise<StoreConnectionInfoWithConnectionOptions> {
-    if (!savedConnectionInfo.connectionModel) {
-      throw new Error(
-        'The connectionModel object is missing in saved connection info.'
+  async loadSavedConnections(): Promise<void> {
+    const globalAndWorkspaceConnections = {
+      ...this._storageController.get(
+        StorageVariables.GLOBAL_SAVED_CONNECTIONS,
+        StorageLocation.GLOBAL
+      ),
+      ...this._storageController.get(
+        StorageVariables.WORKSPACE_SAVED_CONNECTIONS,
+        StorageLocation.WORKSPACE
+      ),
+    };
+
+    let connectionsDidChange = false;
+    const connectionsWithSecretsInKeytar: Array<{
+      connectionId: string;
+      connectionName: string;
+    }> = [];
+    const totalConnectionEntries = Object.entries(
+      globalAndWorkspaceConnections
+    );
+    for (const [connectionId, connectionInfo] of totalConnectionEntries) {
+      const connectionInfoWithSecret = await this._getConnectionInfoWithSecrets(
+        connectionInfo
+      );
+
+      if (connectionInfoWithSecret) {
+        connectionsDidChange = true;
+        this._connections[connectionId] = connectionInfoWithSecret;
+
+        if (
+          connectionInfoWithSecret.secretStorageLocation ===
+          SecretStorageLocation.Keytar
+        ) {
+          connectionsWithSecretsInKeytar.push({
+            connectionId,
+            connectionName: connectionInfo.name,
+          });
+        }
+      }
+    }
```

Comment:
The previous code here was parallelizing the loop with `.map()/Promise.all()`, is there a reason to not do that anymore?
```suggestion
    connectionsWithSecretsInKeytar = (await Promise.all(totalConnectionEntries.map(async ([connectionId, connectionInfo]) => {
      const connectionInfoWithSecret = await this._getConnectionInfoWithSecrets(
        connectionInfo
      );

      if (connectionInfoWithSecret) {
        connectionsDidChange = true;
        this._connections[connectionId] = connectionInfoWithSecret;

        if (
          connectionInfoWithSecret.secretStorageLocation ===
          SecretStorageLocation.Keytar
        ) {
          return {
            connectionId,
            connectionName: connectionInfo.name,
          });
        }
      }
    })).filter(Boolean);
```

Reply:
Not really. This might have also been a mistake as I was too focused in getting that emitted event out of the parallel promises. I will move back to the parallel logic again. Thank you.